File: SocialNetwork/src/main/java/SocialNetwork/SocialNetwork/domain/Response/UserDTO.java
package SocialNetwork.SocialNetwork.domain.Response;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class UserDTO {
    private long id;
    private String email;
    private String fullname;
    private String address;
    private String RoleName;
    private String image;
    private Friend friend;
    @Getter
    @Setter
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Friend{
        private long sumUser;
        private List<UserDTO> friends;
    }

}

File: locket_clone/lib/services/auth/data/models/user_dto.dart
class UserDTO {
  final int id;
  final String email;
  final String fullname;
  final String? address;
  final String? image;
  final String? roleName;

  const UserDTO({
    required this.id,
    required this.email,
    required this.fullname,
    this.address,
    this.image,
    this.roleName,
  });

  factory UserDTO.fromJson(Map<String, dynamic> json) => UserDTO(
    id: (json['id'] as num).toInt(),
    email: json['email'] as String,
    fullname: json['fullname'] as String,
    address: json['address'] as String?,
    image: json['image'] as String?,
    roleName: json['roleName'] as String?,
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'email': email,
    'fullname': fullname,
    'address': address,
    'image': image,
    'roleName': roleName,
  };
}
